hooks/useForm.ts

```diff
@@ -47,70 +52,112 @@ export interface Field extends ProvidedField {
   choice: string
   setChoice: Dispatch<SetStateAction<string>>
 }
+/*
+// eslint-disable-next-line
+type Predicate = (x: any) => boolean
+
+const sometimes = (limit: number, predicate: Predicate) => {
+  let x = 0
+  // eslint-disable-next-line
+  function inner(whatever: any) {
+    x += 1
+    if (x < limit && predicate(whatever)) {
+      // eslint-disable-next-line
+      console.log(JSON.stringify(whatever, null, 2))
+    }
+  }
+  return inner
+}
+*/
 
 export function useField(provided: ProvidedField): Field | null {
-  const [$value, $setter] = useState<string>(provided.defaultValue)
+  const {
+    defaultErrorText = '',
+    defaultLabel = '',
+    defaultValue,
+    id,
+    isRadioed,
+    label,
+    options,
+    radioOption: defaultRadioOption,
+    required = true,
+    touched = false,
+    validation,
+    whitespace = WHITESPACE.DEFAULT,
+  } = provided
+  const [$value, $setter] = useState<string>(defaultValue)
   // TODO: tried writing this with nullish coalescing but it yelled and I got tired
   const radioOption =
-    provided &&
-    provided.options &&
-    provided.options[0] &&
-    provided.options[0].value
-  const { whitespace = WHITESPACE.DEFAULT } = provided
+    defaultRadioOption || (options && options[0] && options[0].value)
   const banSpaces = whitespace === WHITESPACE.BANNED
   const trimSpaces = banSpaces || whitespace === WHITESPACE.TRIMMED
   const [$choice, $setChoice] = useState<string>(
-    provided.isRadioed && radioOption ? radioOption : ''
+    isRadioed && radioOption ? radioOption : ''
   )
   const [$disabled, $setDisabled] = useState<boolean>(false)
-  const [$valid, $setValid] = useState<boolean>(false)
-  const [$touched, $setTouched] = useState<boolean>(false)
+  const [, $setValid] = useState<boolean>(false)
```

Comment:
I don't think this is necessary at all if it's derivative of the other state, and we're not using the state value.

Reply:
Well we're using the `valid` value downstream from here, just not in the `useEffect` block. https://github.com/iron-fish/website-testnet/pull/129/files#diff-731dce45833d5338e722687a46fcec639ee2c63c0cb87a544aeb7a043d3f0df6R115 I can take a look at removing it but I believe I want both the dynamic `valid` and the ability to programmatically invalidate things downstream.

Reply:
I think the direct test here would be whether removing `$setValid` breaks anything. As it stands, the `valid` that gets put into `$setField` is computed from `$touched` and `$value` and isn't React state, so anything that updates either $touched or $value should cause valid to get recomputed and passed through to `$setField`.